spa/src/components/ContributorRouter.test.tsx

```diff
@@ -0,0 +1,190 @@
+    tree('/contributor/contributions/');
+    await screen.findByTestId('mock-contributor-dashboard');
+    expect(
+      within(screen.getByTestId('mock-track-page-view')).getByTestId('mock-contributor-dashboard')
+    ).toBeInTheDocument();
+  });
+
+  it('displays a tracked ContributorEntry at /contributor/', async () => {
+    tree('/contributor/');
+    await screen.findByTestId('mock-contributor-entry');
+    expect(
+      within(screen.getByTestId('mock-track-page-view')).getByTestId('mock-contributor-entry')
+    ).toBeInTheDocument();
+  });
+
+  it('displays a tracked ContributorVerify at /contributor-verify/', async () => {
+    tree('/contributor-verify/');
+    await screen.findByTestId('mock-contributor-verify');
+    expect(
+      within(screen.getByTestId('mock-track-page-view')).getByTestId('mock-contributor-verify')
+    ).toBeInTheDocument();
+  });
+
+  it('renders nothing when: subdomain is not set + no page data + fetchedPageData = false', () => {
+    useRequestMock.mockImplementation(() => (_: any) => {});
+    tree();
+    expect(screen.queryByTestId('mock-segregated-styles')).not.toBeInTheDocument();
+  });
+
+  describe.each([['CORE'], ['FREE'], ['PLUS']])('For the %s plan', (plan) => {
+    // eslint-disable-next-line jest/valid-title
+    describe.each([true, false])('', (hasDefaultDonationPage) => {
+      it(`${hasDefaultDonationPage ? 'if RP has' : 'if RP does not have'} "default_donation_page" use ${
+        plan !== 'FREE' && hasDefaultDonationPage ? 'CUSTOM' : 'DEFAULT'
+      } style`, () => {
+        const mockCustomData = {
+          styles: {
+            font: 'mock-font'
+          },
+          revenue_program: {
+            default_donation_page: hasDefaultDonationPage ? { id: 'mock-id' } : null,
+            organization: {
+              plan: {
+                name: plan
+              }
+            }
+          }
+        };
+        useRequestMock.mockImplementation(
+          () =>
+            (_: any, { onSuccess }: { onSuccess: ({ data }: { data: any }) => void }) =>
+              onSuccess({ data: mockCustomData })
+        );
+        tree();
+        expect.assertions(1);
+        if (plan !== 'FREE' && hasDefaultDonationPage) {
+          // eslint-disable-next-line jest/no-conditional-expect
+          expect(screen.getByTestId('page-data')).toHaveTextContent(JSON.stringify(mockCustomData));
+        } else {
+          // eslint-disable-next-line jest/no-conditional-expect
+          expect(screen.getByTestId('page-data')).toHaveTextContent('false');
+        }
+      });
+    });
+  });
+
+  describe('useWebFonts', () => {
+    it('calls with undefined', () => {
+      useRequestMock.mockImplementation(() => (_: any) => {});
+      tree();
+      expect(useWebFonts).toHaveBeenCalled();
+      expect(useWebFonts).toHaveBeenLastCalledWith(undefined);
+    });
+
+    it('calls with page style', () => {
```

Comment:
```suggestion
    it('calls with page style once loaded', () => {
```
Not sure the description I'm suggesting is accurate, though.

Reply:
Fixed it